src/effects/EffectManager.h

```diff
@@ -81,13 +81,6 @@ class AUDACITY_DLL_API EffectManager
    EffectManager();
    virtual ~EffectManager();
 
-   //! Here solely for the purpose of Nyquist Workbench until a better solution is devised.
-   /** Register an effect so it can be executed.
-     uEffect is expected to be a self-hosting Nyquist effect */
-   const PluginID & RegisterEffect(std::unique_ptr<EffectPlugin> uEffect);
-   //! Used only by Nyquist Workbench module
-   void UnregisterEffect(const PluginID & ID);
```

Comment:
This commit will break compilation of Nyquist Workbench.  Although we are not maintaining that now, I don’t want to abandon all hope of salvaging interesting ideas from that in the future.

Try to keep it at least compilable if not runnable.  Keep the unregister function.  Let the work bench call the other overload of RegisterPlugin passing null for provider.  PluginDescriptor::Load should allow empty providerID without error.

(Also the simlification of Load properly belongs to this commit, not the previous one.)

Reply:
Probably I misunderstood what you said, so I'll revert those changes (except changes in `PluginManager::Load`), and squash this commit with the previous one